guides/source/caching_with_rails.md

```diff
@@ -512,12 +512,21 @@ class ProductsController < ApplicationController
 end
 ```
 
-### A note on weak ETags
+### Strong v/s Weak ETags
+Weak ETags allow semantically equivalent responses to have the same ETags, even
+if their bodies do not match exactly. This is useful when we don't want the page
+to be regenerated for minor changes in response body.
 
-Etags generated by Rails are weak by default. Weak etags allow semantically equivalent responses to have the same etags, even if their bodies do not match exactly. This is useful when we don't want the page to be regenerated for minor changes in response body. If you absolutely need to generate a strong etag, it can be assigned to the header directly.
+```ruby
+  response.weak_etag = response.body # => W/"618bbc92e2d35ea1945008b42799b0e7" → weak ETag
+```
+
+Strong ETag implies that response should be exactly same and byte by byte identical.
+Useful when doing Range requests within a large video or PDF file. Some CDNs
+support only strong ETags, like Akamai.
 
 ```ruby
```

Comment:
Before this code example, we should add ->

Rails generates weak ETags by default. If you absolutely need to generate a strong ETag, it can be done as follows.

``` ruby
class ProductsController < ApplicationController
  def show
    @product = Product.find(params[:id])
    fresh_when last_modified: @product.published_at.utc, strong_etag: @product
  end
end
```

You can also set the strong ETag directly on the response.

``` ruby
response.strong_etag = response.body # => "618bbc92e2d35ea1945008b42799b0e7"
```